File: homework02/Units/Raider.java
package homework02.Units;

import homework02.Names;

public class Raider extends BaseUnit{

    public Raider(String name, int hp) {
        super(name, hp);
        this.strength=12;
        this.dexterity=14;
        this.viability=10;
        this.intelligence=10;
        this.wisdom=10;
        this.charisma=10;
        this.className="Разбойник"; 

    }

    public Raider() {
        this(Names.getRandName(), 120);
    }

    public void daggerStrike(BaseUnit target){
        float damade=10+(float)this.dexterity/10*(float)this.strength/10*BaseUnit.diceRoll();
        System.out.println(this.name+" удар кинжалом "+target.getName());
        target.getDamage((int)damade);
    }
    
}

File: homework07/Units/Raider.cs
class Raider : Infantry
{
    public Raider(int x, int y) : this(Names.getRandName(), x, y) { } 


    public Raider(string name, int x, int y) : base(name, x, y)
    {
        this.attack=8;
        this.defence=3;
        this.damage[0]=2;
        this.damage[1]=4;
        this.hp=10;
        this.maxHp=10;
        this.speed=6;
        this.className="Разбойник"; 
    }
}